src/prefect/tasks/aws/step_function.py

```diff
@@ -37,7 +40,18 @@ def __init__(
 
         super().__init__(**kwargs)
 
-    def run(self, credentials: dict = None):
+    @defaults_from_attrs(
+        "state_machine_arn",
+        "execution_name",
+        "execution_input",
+    )
+    def run(
+        self,
+        credentials: dict = None,
+        state_machine_arn: str = None,
+        execution_name: str = None,
+        execution_input: str = None,
+    ) -> Dict:
```

Comment:
Does it make sense to make boto_kwargs runtime configurable as well? If you think so, then we can also remove `self` from passing it in later.

Reply:
Sounds good. I was following the pattern in other AWS tasks such as [LambdaCreate](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/tasks/aws/lambda_function.py), [S3Download](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/tasks/aws/s3.py), and [AWSSecretsManager](AWSSecretsManager), which only set `boto_kwargs` through the constructor.